apps/dashboard/shared/hooks/useSectionObserver.tsx

```diff
@@ -9,28 +10,23 @@ interface UseSectionObserverProps {
 }
 
 export const useSectionObserver = ({
-  initialSection,
   headerOffset = 0,
   useWindowScrollTo = false,
 }: UseSectionObserverProps) => {
-  const [activeSection, setActiveSection] = useState<string | null>(
-    initialSection ?? null,
-  );
-  const isScrollingRef = useRef<boolean>(false);
+  const { activeSection, updateActiveSection } = useDaoPageInteraction();
   const hasScrolledRef = useRef<boolean>(false);
 
   useEffect(() => {
     const handleSectionChange = (event: Event) => {
-      if (isScrollingRef.current) return;
       const customEvent = event as CustomEvent<string>;
-      setActiveSection(customEvent.detail);
+      updateActiveSection(customEvent.detail, { source: "event" });
     };
 
     window.addEventListener("sectionInView", handleSectionChange);
     return () => {
       window.removeEventListener("sectionInView", handleSectionChange);
```

Comment:
Potential closure issue: `activeSection` in the setTimeout may reference a stale value from when the timeout was created, not the current active section when it executes after 1000ms. Consider capturing the current section ID or removing this parameter since it's updating programmatic state end.